File: VB/SchedulerLocalizationExample_VB/Form1.vb
Imports DevExpress.XtraScheduler
Public Class Form1
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.SchedulerStorage1.Resources.Items.Add(1, "Resource One")
        Me.SchedulerStorage1.Resources.Items.Add(2, "Resource Two")
        Me.SchedulerControl1.GroupType = DevExpress.XtraScheduler.SchedulerGroupType.Resource
    End Sub


    Private Sub SchedulerControl1_EditAppointmentFormShowing(sender As Object, e As AppointmentFormEventArgs) Handles SchedulerControl1.EditAppointmentFormShowing
        Dim scheduler As DevExpress.XtraScheduler.SchedulerControl = CType(sender, DevExpress.XtraScheduler.SchedulerControl)
        Dim curCulture As System.Globalization.CultureInfo = System.Threading.Thread.CurrentThread.CurrentUICulture
        System.Threading.Thread.CurrentThread.CurrentUICulture = New System.Globalization.CultureInfo("de")

        Dim form As SchedulerLocalizationExample_VB.OutlookAppointmentForm = New SchedulerLocalizationExample_VB.OutlookAppointmentForm(scheduler, e.Appointment, e.OpenRecurrenceForm)
        Try
            e.DialogResult = form.ShowDialog
            e.Handled = True
        Finally
            form.Dispose()
        End Try
        System.Threading.Thread.CurrentThread.CurrentUICulture = curCulture

    End Sub
End Class


'#Region "#MySchedulerLocalizer"
Friend Class MySchedulerLocalizer
    Inherits DevExpress.XtraScheduler.Localization.SchedulerLocalizer

    Public Overrides Function GetLocalizedString(ByVal id As DevExpress.XtraScheduler.Localization.SchedulerStringId) As String
        If id = DevExpress.XtraScheduler.Localization.SchedulerStringId.DateTimeAutoFormat_WithoutYear Then
            Return MyBase.GetLocalizedString(id)
        End If
        Return System.Enum.GetName(GetType(DevExpress.XtraScheduler.Localization.SchedulerStringId), id)
    End Function
End Class
'#End Region ' #MySchedulerLocalizer

'#Region "#MyDateNavigatorLocalizer"
Public Class MyDateNavigatorLocalizer
    Inherits DevExpress.XtraEditors.Controls.Localizer

    Public Overrides Function GetLocalizedString(ByVal id As DevExpress.XtraEditors.Controls.StringId) As String
        If id = DevExpress.XtraEditors.Controls.StringId.DateEditToday Then
            Return "Heute"
        End If
        If id = DevExpress.XtraEditors.Controls.StringId.DateEditClear Then
            Return "Löschen"
        End If
        Return MyBase.GetLocalizedString(id)
    End Function
End Class
'#End Region ' #MyDateNavigatorLocalizer

File: CS/SchedulerLocalizerExample/Form1.cs
using System;

namespace SchedulerStringIdExample {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        public Form1() {
            #region #ProgramRunLocalizer
            DevExpress.XtraScheduler.Localization.SchedulerLocalizer.Active = new MySchedulerLocalizer();
            DevExpress.XtraEditors.Controls.Localizer.Active = new MyDateNavigatorLocalizer();
            #endregion #ProgramRunLocalizer

            InitializeComponent();
            this.schedulerStorage1.Resources.Items.Add(1, "Resource One");
            this.schedulerStorage1.Resources.Items.Add(2, "Resource Two");
            this.schedulerControl1.GroupType = DevExpress.XtraScheduler.SchedulerGroupType.Resource;
        }

        private void schedulerControl1_EditAppointmentFormShowing(object sender, DevExpress.XtraScheduler.AppointmentFormEventArgs e) {
            DevExpress.XtraScheduler.SchedulerControl scheduler = ((DevExpress.XtraScheduler.SchedulerControl)(sender));
            System.Globalization.CultureInfo curCulture = System.Threading.Thread.CurrentThread.CurrentUICulture;
            System.Threading.Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo("de");
            SchedulerStringIdExample.OutlookAppointmentForm form = new SchedulerStringIdExample.OutlookAppointmentForm(scheduler, e.Appointment, e.OpenRecurrenceForm);
            try {
                e.DialogResult = form.ShowDialog();
                e.Handled = true;
            }
            finally {
                form.Dispose();
            }
            System.Threading.Thread.CurrentThread.CurrentUICulture = curCulture;
        }

    }
    #region #MySchedulerLocalizer
    class MySchedulerLocalizer : DevExpress.XtraScheduler.Localization.SchedulerLocalizer {
        public override string GetLocalizedString(DevExpress.XtraScheduler.Localization.SchedulerStringId id) {
            if (id == DevExpress.XtraScheduler.Localization.SchedulerStringId.DateTimeAutoFormat_WithoutYear) {
                return base.GetLocalizedString(id);
            }
            return Enum.GetName(typeof(DevExpress.XtraScheduler.Localization.SchedulerStringId), id);
        }
    }
    #endregion #MySchedulerLocalizer

    #region #MyDateNavigatorLocalizer
    public class MyDateNavigatorLocalizer : DevExpress.XtraEditors.Controls.Localizer {
        public override string GetLocalizedString(DevExpress.XtraEditors.Controls.StringId id) {
            if (id == DevExpress.XtraEditors.Controls.StringId.DateEditToday)
                return "Heute";
            if (id == DevExpress.XtraEditors.Controls.StringId.DateEditClear)
                return "Löschen";
            return base.GetLocalizedString(id);
        }
    }
    #endregion #MyDateNavigatorLocalizer
}
